.github/workflows/build-test.yaml

```
@@ -15,6 +15,9 @@ on:
   # Allows triggering this workflow manually from the Action tab
   workflow_dispatch:
 
+permissions:
+  read-all
```

Do you really need read-all? I am not sure if read would be sufficient.
```suggestion
  contents:
    read
```

---

You are right. I missed the following in github's doc:

> If you specify the access for any of these permissions, all of those that are not specified are set to none.